### PR TITLE
Test `is_open` call (on-slave plugin) for many SRs

### PR DESCRIPTION
**Pull request by Wescoeur**
Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

### CODE REVIEW

File: tests/storage/storage.py [hunk 1]
```diff
@@ -43,3 +44,31 @@ def live_storage_migration_then_come_back(vm, prov_host, prov_sr, dest_host, des
     wait_for(lambda: vm.all_vdis_on_sr(prov_sr), "Wait for all VDIs back on provenance SR")
     wait_for(lambda: vm.is_running_on_host(prov_host), "Wait for VM to be running on provenance host")
     vm.shutdown(verify=True)
+
+def vdi_is_open(vdi):
```
Comment (stormi) [inline tests/storage/storage.py:48]: Should be named with an action as it does things.

For example this?
```suggestion
def call_vdi_is_open(vdi):
```

File: tests/storage/xfs/test_xfs_sr.py [hunk 2]
```diff
@@ -39,6 +40,10 @@ def test_create_and_destroy_sr(self, sr_disk, host_with_xfsprogs):
         vm.destroy(verify=True)
         sr.destroy(verify=True)
 
+@pytest.mark.simple_vdi
+def test_xfs_sr_vdi_is_open(vdi_on_xfs_sr):
+    vdi_is_open(vdi_on_xfs_sr)
+
```
Comment (stormi) [inline tests/storage/xfs/test_xfs_sr.py:46]: Why not make it a method of `TestXFSSR`?

File: tests/storage/storage.py (hunk 1 above)
Comment (Wescoeur) [reply tests/storage/storage.py:48]: I modified the function, now a boolean is returned. So I'm not sure we should rename it.

File: tests/storage/xfs/test_xfs_sr.py (hunk 2 above)
Comment (Wescoeur) [reply tests/storage/xfs/test_xfs_sr.py]: Many helpers (Test*SR) use a vm fixture (useless for is_open) like here:
```python
@pytest.mark.usefixtures("nfs_sr", "vm_on_nfs_sr")
class TestNFSSR:
```

I prefer to keep this new test outside these classes.

File: lib/sr.py [hunk 3]
```diff
@@ -130,10 +131,19 @@ def main_host(self):
...
         return self.pool.master.xe('sr-param-get', {'uuid': self.uuid, 'param-name': 'content-type'})
 
     def is_shared(self):
         if self._is_shared is None:
             self._is_shared = self.pool.master.xe('sr-param-get', {'uuid': self.uuid, 'param-name': 'shared'})
         return self._is_shared
+
+    def create_vdi(self, name_label, virtual_size=64):
+        vdi_uuid = self.pool.master.xe('vdi-create', {
+            'name-label': name_label, 'virtual-size': str(virtual_size), 'sr-uuid': self.uuid
```
Comment (stormi) [inline lib/sr.py:147]: It could be good to add the name prefix we already use for various other objects to the name-label.

**Review by gduperrey — APPROVED** [review]

**Comment by stormi** [comment]
We'll wait for the corresponding fix to be released to users before merging this.

**Review by stormi — APPROVED** [review]

**Pull request merged**